packages/ocaml-canvas/ocaml-canvas.1.0.0/opam

```diff
@@ -0,0 +1,67 @@
+opam-version: "2.0"
+synopsis: "The OCaml-Canvas library"
+description: """
+The OCaml-Canvas library provides a portable Canvas for OCaml.
+"""
+authors: ["David Declerck <david.declerck@ocamlpro.com>"]
+maintainer: ["David Declerck <david.declerck@ocamlpro.com>"]
+license: "LGPL-2.1-or-later WITH OCaml-LGPL-linking-exception"
+dev-repo: "git+https://github.com/OCamlPro/ocaml-canvas.git"
+bug-reports: "https://github.com/OCamlPro/ocaml-canvas/issues"
+homepage: "https://ocamlpro.github.io/ocaml-canvas/"
+doc: "https://ocamlpro.github.io/ocaml-canvas/sphinx"
+depends: [
+  "ocaml"                  { >= "4.03" }
+  "dune"                   { >= "2.2" }
+  "dune-configurator"      { >= "1.11" }
+  "react"                  { >= "1.0" }
+  "odoc"                   { >= "1.4" & with-doc }
+
+  "conf-freetype"          { os != "macos" & os != "win32" }
+  "conf-libfontconfig"     { os != "macos" & os != "win32" }
+  "conf-xkbcommon"         { os != "macos" & os != "win32" }
+
+  "conf-libxcb"            { os != "macos" & os != "win32" }
+  "conf-libxcb-shm"        { os != "macos" & os != "win32" }
+  "conf-libxcb-image"      { os != "macos" & os != "win32" }
+  "conf-libxcb-xkb"        { os != "macos" & os != "win32" }
+  "conf-libxcb-keysyms"    { os != "macos" & os != "win32" }
+]
+depopts: [
+  "js_of_ocaml"            { >= "3.6" }
+ "conf-wayland-protocols" {os != "macos" & os != "win32"}
+ "conf-libwayland" {os != "macos" & os != "win32"}
+]
```

Comment:
```suggestion
depopts: [
  "js_of_ocaml"
  "conf-wayland-protocols" {os != "macos" & os != "win32"}
  "conf-libwayland" {os != "macos" & os != "win32"}
]
conflicts: [
  "js_of_ocaml" { < "3.6" }
]
```
The behaviour of filters in the `depopts` fields is probably not what you want

Reply:
Not sure. When building my library, I check whether `js_of_ocaml` is available and >= 3.6. If not, then the optional JS parts will simply not be built ; this will not cause a build failure. As I understand, the `conflict` specification will _prevent_ the library from being installed if `js_of_ocaml` is installed and < 3.6, right ?

Reply:
mmh, i just looked your detection code and it looks correct indeed. Fair enough then, I'll remove my change and merge this PR. But be really careful, this feature really gives you the gun to shoot yourself in the foot.